src/groupeddataframe/grouping.jl

```diff
@@ -238,76 +238,15 @@ function _groupvar_idx(gd::GroupedDataFrame, name::Symbol, strict::Bool)
-    groupvalues(gd::GroupedDataFrame, i, col)
-
-Get the values of the grouping columns for `gd`, optionally specifying the group
-index `i` and grouping column `col`. `col` may be an integer index (of the
-grouping columns passed to [`groupby`](@ref)) or the column name itself. The
-optional first argument to the first two forms specifies the return type, which
-may be `NamedTuple` (default) or `Tuple`.
-
-# Returns
+# Get values of grouping columns for all groups
+_groupvalues(gd::GroupedDataFrame) = [_groupvalues(gd, i) for i in 1:length(gd)]
 
-Iterator over `Tuple`s/`NamedTuple`s, a single `Tuple`/`NamedTuple`, or a
-grouping column value.
-
-# Examples
+# Get values of grouping columns for single group
+_groupvalues(gd::GroupedDataFrame, i::Integer) = gd.parent[gd.idx[gd.starts[i]], gd.cols]
 
-```jldoctest
-julia> df = DataFrame(a = repeat([:foo, :bar, :baz], outer=[4]),
-                      b = repeat([2, 1], outer=[6]),
-                      c = 1:12);
-
-julia> gd = groupby(df, [:a, :b])
-GroupedDataFrame with 6 groups based on keys: a, b
-First Group (2 rows): a = :foo, b = 2
-│ Row │ a      │ b     │ c     │
-│     │ Symbol │ Int64 │ Int64 │
-├─────┼────────┼───────┼───────┤
-│ 1   │ foo    │ 2     │ 1     │
-│ 2   │ foo    │ 2     │ 7     │
-⋮
-Last Group (2 rows): a = :baz, b = 1
-│ Row │ a      │ b     │ c     │
-│     │ Symbol │ Int64 │ Int64 │
-├─────┼────────┼───────┼───────┤
-│ 1   │ baz    │ 1     │ 6     │
-│ 2   │ baz    │ 1     │ 12    │
-
-julia> collect(groupvalues(gd))
-6-element Array{NamedTuple{(:a, :b),Tuple{Symbol,Int64}},1}:
- (a = :foo, b = 2)
- (a = :bar, b = 1)
- (a = :baz, b = 2)
- (a = :foo, b = 1)
- (a = :bar, b = 2)
- (a = :baz, b = 1)
-
-julia> groupvalues(gd, 2)
-(a = :bar, b = 1)
-
-julia> groupvalues(gd, 2, 1)
-:bar
-
-julia> groupvalues(gd, 2, :a)
-:bar
-
-julia> groupvalues(Tuple, gd, 2)
-(:bar, 1)
-```
-"""
-groupvalues(T::Type, gd::GroupedDataFrame) = (groupvalues(T, gd, i) for i in 1:length(gd))
-groupvalues(gd::GroupedDataFrame) = groupvalues(NamedTuple, gd)
-
-groupvalues(::Type{Tuple}, gd::GroupedDataFrame, i::Integer) = Tuple(gd.parent[gd.idx[gd.starts[i]], gd.cols])
-groupvalues(::Type{NamedTuple}, gd::GroupedDataFrame, i::Integer) = NamedTuple{Tuple(groupvars(gd))}(groupvalues(Tuple, gd, i))
-groupvalues(gd::GroupedDataFrame, i::Integer) = groupvalues(NamedTuple, gd, i)
-
-groupvalues(gd::GroupedDataFrame, i::Integer, col::Integer) = gd.parent[gd.idx[gd.starts[i]], gd.cols[col]]
-groupvalues(gd::GroupedDataFrame, i::Integer, col::Symbol) = groupvalues(gd, i, _groupvar_idx(gd, col, true))
+# Get values of single grouping column for single group
+_groupvalues(gd::GroupedDataFrame, i::Integer, col::Integer) = gd.parent[gd.idx[gd.starts[i]], gd.cols[col]]
+_groupvalues(gd::GroupedDataFrame, i::Integer, col::Symbol) = _groupvalues(gd, i, _groupvar_idx(gd, col, true))
 
 # Eltypes of the grouping columns
 _grouptypes(gd::GroupedDataFrame) = Tuple(eltype(gd.parent[!, c]) for c in gd.cols)
```

Comment:
This isn't used anymore it seems.

Reply:
This are used in `getindex` and `getproperty` of `GroupKey`, I think.

Reply:
I do not see the usage of this function in this PR.

Reply:
Oh, I though this was about `_groupvalues` methods. I've removed `_grouptypes`.